conf/imx_hab4/sign-spl.sh

```diff
@@ -0,0 +1,140 @@
+#!/bin/sh
+#
+# Copyright (c) 2019 Foundries.io
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+CST_BINARY="cst"
+CSF_TEMPLATE="u-boot-spl-sign.csf-template"
+SPL_FILE="SPL"
+KEY_DIR="."
+DCD_CLEAR=0
+DISPLAY_USAGE=0
+
+function parse_args()
+{
+    while [ ${#} -gt 0 ]
+    do
+        case ${1} in
+        --cst)
+            CST_BINARY=${2}
+            shift
+            shift
+            ;;
+        --csf-template)
+            CSF_TEMPLATE=${2}
+            shift
+            shift
+            ;;
+        --spl)
+            SPL_FILE=${2}
+            shift
+            shift
+            ;;
+        --key-dir)
+            KEY_DIR=${2}
+            shift
+            shift
+            ;;
+        --mfgtools)
+            DCD_CLEAR=1
+            shift
+            ;;
+        -h)
+            DISPLAY_USAGE=1
+            shift
+            ;;
+        --help)
+            DISPLAY_USAGE=1
+            shift
+            ;;
+        *)
+            shift
+            ;;
+        esac
+    done
+}
+
+parse_args "$@"
+
+if [ "${DISPLAY_USAGE}" == "1" ]; then
+    echo "usage for: ${0}"
+    echo "  --cst: set cst binary path/filename   [default: ${CST_BINARY}]"
+    echo "  --csf-template: set CSF template file [default: ${CSF_TEMPLATE}]"
+    echo "  --spl: set filename for SPL binary    [default: ${SPL_FILE}]"
+    echo "  --key-dir: location for key files     [default: ${KEY_DIR}]"
+    echo "  --mfgtools: turn on clear / restore DCD addr for SPD SPL binary"
```

Comment:
I should change the help text here.  It doesn't "restore" anything anymore.  It modifies a copy of the SPL binary (SPL.mod)

Reply:
mfgtools is confusing as what we want with this option is the possibility of loading a signed SPL over SDP

Reply:
yeah probably better name for the flag required here: --fix-sdp-dcd or something.

Reply:
Woops missed this.  So was `--fix-sdp-dcd` a good replacement for `--mfgtools`?

Reply:
Ok fixed in latest version